subtests/docker_cli/run_cgroup_parent/run_cgroup_parent.py

```diff
@@ -0,0 +1,187 @@
+
+            # Must find at least one matching cgroup line
+            if not found_match:
+                raise DockerTestFail("No output from cgroups")
+        else:
+            self.failif_ne(stdout, '', "unexpected output on stdout")
+
+    def _read_cid(self):
+        """
+        Read container ID from --cidfile file, so we can replace {cid}
+        in format string and do an exact match on expected values.
+        """
+        cid = open(self.sub_stuff['cidfile'], 'rb').read().strip()
+        if len(cid) < 12:
+            raise DockerTestError("bad cid (length < 12) in --cidfile")
+        return cid
+
+    def cleanup(self):
+        """
+        Remove stray cgroups
+        """
+        cgroups_dir = '/sys/fs/cgroup'
+        try:
+            rand1 = self.sub_stuff['rand1']
+            rand2 = self.sub_stuff['rand2']
+        except KeyError:                # Test never ran
+            return
+
+        for ent in os.listdir(cgroups_dir):
+            # eg /sys/fs/cgroup/memory
+            controller_dir = os.path.join(cgroups_dir, ent)
+            if os.path.isdir(controller_dir):
+                # eg /sys/fs/cgroup/memory/foo.slice
+                rand1_dir = os.path.join(controller_dir, "%s.slice" % rand1)
+                if os.path.isdir(rand1_dir):
+                    # eg /sys/fs/cgroup/memory/foo.slice/foo-bar.slice
+                    rand2_dir = os.path.join(rand1_dir,
+                                             "%s-%s.slice" % (rand1, rand2))
+                    # Note that this is special cgroupfs magic! rmdir
+                    # succeeds even though ls lists "files" within.
+                    if os.path.isdir(rand2_dir):
+                        os.rmdir(rand2_dir)
+                    os.rmdir(rand1_dir)
+
+
+class run_cgroup_parent_invalid_name(run_cgroup_parent_base):
+    """
+    Invalid value for --cgroup-parent option
+    """
+
+    def initialize(self):
+        super(run_cgroup_parent_invalid_name, self).initialize()
+        self._setup("/{rand1}")
+        self._expect(stderr="Error response from daemon:"
+                     " Cannot start container {cid}: [8] System error:"
+                     " Invalid slice name /{rand1}")
+
+
+class run_cgroup_parent_path(run_cgroup_parent_base):
+    """
+    Perfectly valid slice. In docker < 1.9.1-25.el7 this produced
+    weird results.
+    """
+
+    def initialize(self):
+        super(run_cgroup_parent_path, self).initialize()
+        self._setup("{rand1}.slice")
+        self._expect(path="/{rand1}.slice/docker-{cid}.scope")
+
+
+class run_cgroup_parent_path_with_hyphens(run_cgroup_parent_base):
+    """
+    Hyphens result in a more complicated path.
+    """
+
+    def initialize(self):
+        super(run_cgroup_parent_path_with_hyphens, self).initialize()
+        self._setup("{rand1}-{rand2}.slice")
+        self._expect(path="/{rand1}.slice/{rand1}-{rand2}.slice"
+                     "/docker-{cid}.scope")
```

Comment:
hahaha, I had a feeling starting fresh would be a lot easier.  That other cgroups test is quite crusty.